tests/core/math/test_vector2.h

```diff
@@ -0,0 +1,67 @@
+/*************************************************************************/
+/*  test_vector2.h                                                       */
+/*************************************************************************/
+/*                       This file is part of:                           */
+/*                           GODOT ENGINE                                */
+/*                      https://godotengine.org                          */
+/*************************************************************************/
+/* Copyright (c) 2007-2021 Juan Linietsky, Ariel Manzur.                 */
+/* Copyright (c) 2014-2021 Godot Engine contributors (cf. AUTHORS.md).   */
+/*                                                                       */
+/* Permission is hereby granted, free of charge, to any person obtaining */
+/* a copy of this software and associated documentation files (the       */
+/* "Software"), to deal in the Software without restriction, including   */
+/* without limitation the rights to use, copy, modify, merge, publish,   */
+/* distribute, sublicense, and/or sell copies of the Software, and to    */
+/* permit persons to whom the Software is furnished to do so, subject to */
+/* the following conditions:                                             */
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#ifndef GODOT_TEST_VECTOR2_H
+#define GODOT_TEST_VECTOR2_H
+
+#include "core/math/Vector2.h"
+
+#include "tests/test_macros.h"
+
+namespace TestVector2 {
+
+TEST_CASE("[Vector2] Constructor methods") {
+	const Vector2 vector2 = Vector2(1.2, 2.1);
+	const Vector2 vector2_copy = Vector2(vector2);
+
+	CHECK_MESSAGE(
+			vector2 == vector2_copy, "Vector2s created with the same values but different methods should be equal.");
+}
+
+TEST_CASE("[Vector2] Transpose methods") {
+	const Vector2 vector2 = Vector2(1.2, 2.1);
+	const Vector2 xy = vector2.get_xy();
+	const Vector2 yx = vector2.get_yx();
+
+	CHECK_MESSAGE(
```

Comment:
How can I check the status of these tests? Are they even running?

Reply:
You have to add them to `test_main.cpp`: https://github.com/godotengine/godot/blob/master/tests/test_main.cpp

Then they'll be run in CI, and you can execute tests locally by compiling using the scons flag `tests=yes` then executing the binary with the flag `--test`.

Reply:
Alternatively, I already have a PR that adds tests for Vector2/Vector2i/Vector3/Vector3i, so you can just wait for those to be merged: https://github.com/godotengine/godot/pull/55041

Reply:
I will add them manually for now so we don't have to deadlock this PR